nibabel/cmdline/diff.py

```diff
@@ -125,6 +140,80 @@ def get_data_diff(files):
     return md5sums
 
 
+def get_data_diff(files, max_abs=0, max_rel=0):
+    """Get difference between data
+
+    Parameters
+    ----------
+    files: list of (str or ndarray)
+      If list of strings is provided -- they must be existing file names
+    max_abs: float, optional
+      Maximal absolute difference to tolerate.
+    max_rel: float, optional
+      Maximal relative (`abs(diff)/mean(diff)`) difference to tolerate.
+      If `max_abs` is specified, then those data points with lesser than that
+      absolute difference, are not considered for relative difference testing
+
+    Returns
+    -------
+    diffs: OrderedDict
+        An ordered dict with a record per each file which has differences with other files subsequent detected.
+        Each record is a list of difference records, one per each file pair. Each difference record is an Ordered
+        Dict with possible keys 'abs' or 'rel' showing maximal absolute or relative differences in the file 
+        or record ('CMP': 'incompat') if file shapes are incompatible.
+    """
+    
+    # we are doomed to keep them in RAM now
+    data = [f if isinstance(f, np.ndarray) else nib.load(f).get_fdata() for f in files]
```

Comment:
Note that `get_fdata()` returns float64 arrays by default. If you're hoping to keep things moderately compact, you could use `get_fdata(dtype=np.float32)`. The precision loss should not affect equivalent files, and should be in the noise for any plausible MRI data.